java/code/src/com/redhat/rhn/manager/appstreams/AppStreamsManager.java

```diff
@@ -182,33 +183,9 @@ public static AppStream findAppStream(Long channelId, String name, String stream
      */
     public static void cloneAppStreams(Channel to, Channel from) {
         if (from.isModular()) {
-            AppStreamsManager.listChannelAppStreams(from.getId()).forEach(appStream -> {
-                AppStream a = new AppStream();
-                a.setChannel(to);
-                a.setName(appStream.getName());
-                a.setStream(appStream.getStream());
-                a.setVersion(appStream.getVersion());
-                a.setContext(appStream.getContext());
-                a.setArch(appStream.getArch());
-                cloneArtifacts(a, appStream);
-                HibernateFactory.getSession().persist(a);
```

Comment:
Did you try moving the persist/save to outside the stream and see if it makes any difference in the performance? I'm asking this because I really dislike psql to have any logic (hard to debug and maintain).
However, if it is the only possibility to have good performance, then we need to use it.

Reply:
got it. I thought the 30 seconds were with the persist/save  calls outside the stream. Thank you for clarification